### PR TITLE
[Merged by Bors] - Use build rbac resources from operator-rs 

### PR DESCRIPTION
**Pull request by razvan**
Tests successful on OS 4.10

```
--- PASS: kuttl (2911.39s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/mount-dags-configmap_airflow-latest-2.4.1-stackable23.4.0-rc2_openshift-false (190.83s)
        --- PASS: kuttl/harness/smoke_airflow-2.4.1-stackable23.4.0-rc3_openshift-false (231.09s)
        --- PASS: kuttl/harness/smoke_airflow-2.4.1-stackable23.4.0-rc2_openshift-false (208.10s)
        --- PASS: kuttl/harness/logging_airflow-2.2.4-stackable23.4.0-rc2_openshift-false (263.57s)
        --- PASS: kuttl/harness/smoke_airflow-2.2.5-stackable23.4.0-rc3_openshift-false (231.60s)
        --- PASS: kuttl/harness/smoke_airflow-2.2.5-stackable23.4.0-rc2_openshift-false (297.47s)
        --- PASS: kuttl/harness/smoke_airflow-2.2.4-stackable23.4.0-rc3_openshift-false (221.94s)
        --- PASS: kuttl/harness/smoke_airflow-2.2.4-stackable23.4.0-rc2_openshift-false (223.27s)
        --- PASS: kuttl/harness/cluster-operation_airflow-latest-2.4.1-stackable23.4.0-rc3_openshift-false (233.68s)
        --- PASS: kuttl/harness/logging_airflow-2.4.1-stackable23.4.0-rc3_openshift-false (170.48s)
        --- PASS: kuttl/harness/logging_airflow-2.4.1-stackable23.4.0-rc2_openshift-false (198.20s)
        --- PASS: kuttl/harness/logging_airflow-2.2.5-stackable23.4.0-rc3_openshift-false (218.00s)
        --- PASS: kuttl/harness/logging_airflow-2.2.5-stackable23.4.0-rc2_openshift-false (188.41s)
        --- PASS: kuttl/harness/logging_airflow-2.2.4-stackable23.4.0-rc3_openshift-false (325.02s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.4.1-stackable23.4.0-rc2_ldap-authentication-insecure-tls_openshift-false (254.43s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.4.1-stackable23.4.0-rc3_ldap-authentication-server-verification-tls_openshift-false (212.81s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.4.1-stackable23.4.0-rc3_ldap-authentication-no-tls_openshift-false (208.23s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.4.1-stackable23.4.0-rc3_ldap-authentication-insecure-tls_openshift-false (208.31s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.4.1-stackable23.4.0-rc2_ldap-authentication-server-verification-tls_openshift-false (175.93s)
        --- PASS: kuttl/harness/resources_airflow-latest-2.4.1-stackable23.4.0-rc2_openshift-false (148.40s)
        --- PASS: kuttl/harness/ldap_airflow-latest-2.4.1-stackable23.4.0-rc2_ldap-authentication-no-tls_openshift-false (196.71s)
        --- PASS: kuttl/harness/resources_airflow-latest-2.4.1-stackable23.4.0-rc3_openshift-false (183.10s)
        --- PASS: kuttl/harness/mount-dags-gitsync_airflow-latest-2.4.1-stackable23.4.0-rc3_openshift-false (229.54s)
        --- PASS: kuttl/harness/mount-dags-configmap_airflow-latest-2.4.1-stackable23.4.0-rc3_openshift-false (198.03s)
        --- PASS: kuttl/harness/orphaned-resources_airflow-latest-2.4.1-stackable23.4.0-rc3_openshift-false (205.12s)
        --- PASS: kuttl/harness/orphaned-resources_airflow-latest-2.4.1-stackable23.4.0-rc2_openshift-false (212.96s)
```

### CODE REVIEW

File: rust/operator-binary/src/airflow_controller.rs [hunk 1]
```diff
@@ -138,6 +139,10 @@ pub enum Error {
         name: String,
         source: stackable_operator::error::Error,
     },
+    #[snafu(display("failed to build RBAC objects: {source}"))]
+    BuildRBACObjects {
+        source: stackable_operator::error::Error,
+    },
```
Comment (maltesander) [inline rust/operator-binary/src/airflow_controller.rs:145]:
```suggestion
    #[snafu(display("failed to build RBAC objects"))]
    BuildRBACObjects {
        source: stackable_operator::error::Error,
    },
```

File: rust/operator-binary/src/airflow_controller.rs [hunk 2]
```diff
@@ -1,11 +1,11 @@
 //! Ensures that `Pod`s are configured and running for each [`AirflowCluster`]
 use crate::config::{self, PYTHON_IMPORTS};
+use crate::controller_commons;
 use crate::controller_commons::{CONFIG_VOLUME_NAME, LOG_CONFIG_VOLUME_NAME, LOG_VOLUME_NAME};
 use crate::product_logging::{
     extend_config_map_with_log_config, resolve_vector_aggregator_address,
 };
```
Comment (maltesander) [inline rust/operator-binary/src/airflow_controller.rs:7]:
```suggestion
use crate::controller_commons::{
    self, CONFIG_VOLUME_NAME, LOG_CONFIG_VOLUME_NAME, LOG_VOLUME_NAME,
};
use crate::product_logging::{
    extend_config_map_with_log_config, resolve_vector_aggregator_address,
};
```

File: rust/operator-binary/src/airflow_controller.rs [hunk 3]
```diff
@@ -306,6 +293,26 @@ pub async fn reconcile_airflow(airflow: Arc<AirflowCluster>, ctx: Arc<Ctx>) -> R
...
+        .await
+        .with_context(|_| ApplyServiceAccountSnafu {
+            name: rbac_sa.name_unchecked(),
+        })?;
+    cluster_resources
+        .add(client, rbac_rolebinding.clone())
+        .await
+        .with_context(|_| ApplyRoleBindingSnafu {
+            name: rbac_rolebinding.name_unchecked(),
+        })?;
```
Comment (maltesander) [inline rust/operator-binary/src/airflow_controller.rs:314]: You can remove the name from the error (not used in the display string and will be propagated anyways).
This avoids the clones as well.
```suggestion
    let rbac_sa = cluster_resources
        .add(client, rbac_sa)
        .await
        .context(ApplyServiceAccountSnafu)?;
    cluster_resources
        .add(client, rbac_rolebinding)
        .await
        .context(ApplyRoleBindingSnafu)?;
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Minor things. LGTM otherwise.

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**